client/src/utils/ajax.ts

```diff
@@ -207,6 +208,13 @@ export function postWebhookToken(): Promise<void> {
   return post('/user/webhook-token', {});
 }
 
+export function postSaveChallenge(body: {
+  id: string;
+  files: ChallengeFiles;
```

Comment:
Should make use of `parseApiResponseToClientUser` function somewhere here - also need to change function to correctly parse `savedChallenges` field from server.